solutions/cses-1161.mdx

```diff
@@ -55,4 +56,42 @@ int main(){
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class StickDivision {
+	public static void main(String[] args) throws IOException {
+		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(in.readLine());
+
+		// with our algorithm the variable x will not be used at all
+		// since it is guaranteed by problem constraints that all values add up to x
+		int x = Integer.parseInt(st.nextToken());
+		int n = Integer.parseInt(st.nextToken());
+		Queue<Integer> sticks = new PriorityQueue<Integer>();
+		st = new StringTokenizer(in.readLine());
+		for (int i = 0; i < n; i++)
+			sticks.add(Integer.valueOf(st.nextToken()));
+
+		long costs = 0;
+		// we keep combining two smallest sticks a and b
+		// until there is only one stick remaining, which will have the length x
+		while (sticks.size() > 1) {
+			int a = sticks.poll();
```

Comment:
optional change, do .remove() instead of .pol() because it is more intuitive and easier to read and understand

the only different is that poll gets null if empty, remove throws an error if empty

Reply:
i've always used poll?

Reply:
na I usually use poll() for queue but I can certainly change it to remove()